src/main/zapHomeFiles/hud/serviceworker.js

```diff
@@ -167,6 +183,18 @@ webSocket.onmessage = function (event) {
 		utils.log(LOG_DEBUG, 'serviceworker.webSocket.onmessage', jevent['event.publisher']);
 		var ev = new CustomEvent(jevent['event.publisher'], {detail: jevent});
 		self.dispatchEvent(ev);
+	} else if ('id' in jevent && 'response' in jevent) {
+		let pFunctions = webSocketCallbacks[jevent['id']];
+		let response = jevent['response'];
+		if ('code' in response && 'message' in response) {
+			// These always indicate a failure
+			pFunctions.reject(response);
+		} else {
+			pFunctions.resolve(response);
```

Comment:
maybe in the failure case we should wrap the response in an `Error` object and supply the response as event data?

Reply:
yeah, looks like wrapping the response in an Error is the 'standard' way to go.